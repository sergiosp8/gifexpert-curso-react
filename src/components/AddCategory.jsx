import { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('')

  const onInputChange = (e) => {
    setInputValue(e.target.value)
  }

  const onSubmitForm = (e) => {
    e.preventDefault()
    if (inputValue.trim().length <= 1) return
    onNewCategory(inputValue.trim())
    setInputValue('')
  }
  return (
    <form onSubmit={onSubmitForm} aria-label="form">
      <input type="text" placeholder="Buscar Gifs" value={inputValue} onChange={onInputChange} />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
