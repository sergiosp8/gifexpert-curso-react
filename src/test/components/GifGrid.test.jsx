import { render, screen } from '@testing-library/react'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../Hooks/useFetchGifs'

jest.mock('../../Hooks/useFetchGifs')

describe('Pruebas <GifGrid/>', () => {
  useFetchGifs.mockReturnValue({
    images: [],
    isLoading: true
  })
  const category = 'One Punch'
  test('Debe de mostrar el componente correctamente', () => {
    const { container } = render(<GifGrid category={category} />)
    expect(container).toMatchSnapshot()
  })

  test('debe de mostrar el loading inicialmente', () => {
    useFetchGifs.mockReturnValue({
      images: [],
      isLoading: true
    })

    render(<GifGrid category={category} />)

    expect(screen.getByText('Loading...')).toBeTruthy()
  })

  test('debe de mostrar items cuando se cargan las imagenes useFetchGifs', () => {
    const gifs = [
      { id: 'ABC', title: 'Titulo 1', url: 'https://url.com' },
      { id: 'DEF', title: 'Titulo 2', url: 'https://url.com' },
      { id: 'GHI', title: 'Titulo 3', url: 'https://url.com' }
    ]

    useFetchGifs.mockReturnValue({
      images: gifs,
      isLoading: false
    })

    render(<GifGrid category={category} />)

    const imgs = screen.getAllByRole('img')

    expect(imgs.length).toBe(3)
  })
})
