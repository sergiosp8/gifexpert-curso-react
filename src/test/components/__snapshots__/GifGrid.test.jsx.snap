// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifGrid/> Debe de mostrar el componente correctamente 1`] = `
<div>
  <h3>
    One Punch
  </h3>
  <p>
    Loading...
  </p>
  <div
    class="card-grid"
  />
</div>
`;
